### PR TITLE
samples: intel_s1000: I2S audio sample application

### PR DESCRIPTION
**Pull request by sathishkuttan**
This PR adds a sample application for the Intel S1000 board.
Audio can be played out in one of 2 modes
- A sequence of tones generated by the app
- A passthrough mode in which the app forwards the audio from another host connected to S1000 over I2S bus.

### CODE REVIEW

File: samples/boards/intel_s1000_crb/i2s/i2s_app.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+The application uses the following drivers
+- I2S driver
+- Codec driver
+
+Two instances of the I2S driver are used.
+- I2S_1 is used as the output interface to play audio to the onboard codec.
+- I2S_2 is used as the input interface to receive audio from a connected host.
+
+Intel® S1000 is the master on both I2S interfaces.
+The host is a slave on I2S_1 and is expected to send a stereo audio at a
```
Comment (rgundi) [inline samples/boards/intel_s1000_crb/i2s/i2s_app.rst]: Should this be I2S_2?

File: samples/boards/intel_s1000_crb/i2s/prj.conf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CONFIG_LOG=y
+
```
Comment (rgundi) [inline samples/boards/intel_s1000_crb/i2s/prj.conf]: No need to increase the LOG buffer here? We had crashes in the tests folder due to that.

**Review by rgundi — COMMENTED** [review]
Guess you may also need to whitelist this for s1000 in a yaml file.

File: samples/boards/intel_s1000_crb/i2s/i2s_app.rst (hunk 1 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/i2s/i2s_app.rst]: Good catch. Fixed.

**Comment by sathishkuttan** [comment]
> Guess you may also need to whitelist this for s1000 in a yaml file.

What does the yaml file do? I didn't have one for the DMIC sample `samples/boards/intel_s1000_crb/dmic/dmic.rst`

**Comment by rgundi** [comment]
> > Guess you may also need to whitelist this for s1000 in a yaml file.
> 
> What does the yaml file do? I didn't have one for the DMIC sample `samples/boards/intel_s1000_crb/dmic/dmic.rst`

From what i know, in the tests folder, the yaml files get picked up by the sanitycheck for executing the corresponding tests. I am not quite sure about its use in the samples folder. However, i see the other samples in the samples folder do have the yaml file so you may consider having it.

**Comment by sathishkuttan** [comment]
> From what i know, in the tests folder, the yaml files get picked up by the sanitycheck for executing the corresponding tests. I am not quite sure about its use in the samples folder. However, i see the other samples in the samples folder do have the yaml file so you may consider having it.

I see. I'll talk to @nashif later and add the YAML in a separate PR for all sample apps as needed.

File: samples/boards/intel_s1000_crb/i2s/i2s_app.rst [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+Intel® S1000 is the master on both I2S interfaces.
+The host is a slave on the I2S and is expected to send a stereo audio at a
+sampling frequency of 48KHz, 32 bits per sample.
+
+The app can be built in one of two modes and the mode selection is done by
+providing a command line flag to ``cmake``.
+
+1. **Audio Playback from Host**
+
+   In this mode, the audio is played from the host. For example, on linux hosts,
```
Comment (dbkinder) [inline samples/boards/intel_s1000_crb/i2s/i2s_app.rst]: ``Linux`` should be capitalized

**Review by dbkinder — DISMISSED** [review]
one nit in the doc, otherwise LGTM

File: samples/boards/intel_s1000_crb/i2s/i2s_app.rst (hunk 3 above)
Comment (sathishkuttan) [reply samples/boards/intel_s1000_crb/i2s/i2s_app.rst]: Fixed.

**Comment by nashif** [comment]
please add the sample meta-data (sample.yaml):
```yaml
sample:
  name: I2S Audio Sample
tests:
  test:
    platform_whitelist: intel_s1000_crb
    tags: i2s
```


This will allow integration into testing.

**Comment by sathishkuttan** [comment]
> please add the sample meta-data (sample.yaml):

Done. Thanks for the yaml content!

**Review by nashif — APPROVED** [review]

**Pull request merged**